packages/node/src/indexer/dictionary/v1/ethDictionaryV1.ts

```diff
@@ -0,0 +1,260 @@
+): DictionaryV1QueryEntry {
+  const conditions: DictionaryQueryCondition[] = [];
+  appendDsOptions(dsOptions, conditions);
+  if (filter.topics) {
+    for (let i = 0; i < Math.min(filter.topics.length, 4); i++) {
+      const topic = filter.topics[i];
+      if (!topic) {
+        continue;
+      }
+      const field = `topics${i}`;
+
+      if (topic === '!null') {
+        conditions.push({
+          field,
+          value: false as any, // TODO update types to allow boolean
+          matcher: 'isNull',
+        });
+      } else {
+        conditions.push({
+          field,
+          value: eventToTopic(topic),
+          matcher: 'equalTo',
+        });
+      }
+    }
+  }
+  return {
+    entity: 'evmLogs',
+    conditions,
+  };
+}
+
+function callFilterToQueryEntry(
+  filter: EthereumTransactionFilter,
+  dsOptions: SubqlEthereumProcessorOptions | SubqlEthereumProcessorOptions[],
+): DictionaryV1QueryEntry {
+  const conditions: DictionaryQueryCondition[] = [];
+  appendDsOptions(dsOptions, conditions);
+
+  for (const condition of conditions) {
+    if (condition.field === 'address') {
+      condition.field = 'to';
+    }
+  }
+  if (filter.from) {
+    conditions.push({
+      field: 'from',
+      value: filter.from.toLowerCase(),
+      matcher: 'equalTo',
+    });
+  }
+  const optionsAddresses = conditions.find((c) => c.field === 'to');
+  if (!optionsAddresses) {
+    if (filter.to) {
+      conditions.push({
+        field: 'to',
+        value: filter.to.toLowerCase(),
+        matcher: 'equalTo',
+      });
+    } else if (filter.to === null) {
+      conditions.push({
+        field: 'to',
+        value: true as any, // TODO update types to allow boolean
+        matcher: 'isNull',
+      });
+    }
+  } else if (optionsAddresses && (filter.to || filter.to === null)) {
+    logger.warn(
+      `TransactionFilter 'to' conflict with 'address' in data source options`,
+    );
+  }
+
+  if (filter.function) {
+    conditions.push({
+      field: 'func',
+      value: functionToSighash(filter.function),
+      matcher: 'equalTo',
+    });
+  }
+  return {
+    entity: 'evmTransactions',
+    conditions,
+  };
+}
+
+export type GroupedEthereumProjectDs = SubqlDatasource & {
+  groupedOptions?: SubqlEthereumProcessorOptions[];
+};
+
+export function buildDictionaryV1QueryEntries(
+  dataSources: GroupedEthereumProjectDs[],
+): DictionaryV1QueryEntry[] {
+  const queryEntries: DictionaryV1QueryEntry[] = [];
+
+  for (const ds of dataSources) {
+    for (const handler of ds.mapping.handlers) {
+      // No filters, cant use dictionary
+      if (!handler.filter) return [];
+
+      switch (handler.kind) {
+        case EthereumHandlerKind.Block:
+          return [];
+        case EthereumHandlerKind.Call: {
+          const filter = handler.filter as EthereumTransactionFilter;
+          if (
+            filter.from !== undefined ||
+            filter.to !== undefined ||
+            filter.function
```

Comment:
```suggestion
            filter.function !== undefined
```
The other changes from this PR should also be applied https://github.com/subquery/subql-ethereum/commit/81eb55d2978252190242f62a7dcff505c1d0db9c#diff-fcb06f77cc4b787f2ed3c116347e39144575cd2ce8fb07af6b6ea11ef0ac2a05R196